src/router.ts

```diff
@@ -173,12 +179,22 @@ class Router {
           )
         ] = dipDeposit;
       }
-      for (const dip of Object.values(this.dips)) {
-        if (dip.qtyTokens > 0) {
-          openPositionCount++;
-          // TODO await for a response back from the API
-          await this.route(dip);
+      let totalRoutedQty = 0;
+      for (let i = 0; i < MAX_ROUTE_ATTEMPTS; i++) {
+        for (const dip of Object.values(this.dips)) {
+          if (dip.qtyTokens > 0) {
+            openPositionCount++;
+            // Poller will immediately fire after position changes there may be a race to route
+            // See TODO: Delaying Poller or Putting on a timer
+            totalRoutedQty += await this.route(dip);
+          }
+        }
+        if (totalRoutedQty === 0) {
```

Comment:
should this be routing on the last iteration, not total?

Reply:
If we just return whenever we have a non-zero route, because we got a sale to MM, will we also get a rerun due to the balance change of the DIP account. So terminate here and the rerun handles it?

Reply:
Brilliant, yea do nothing after routing a positive amount.. must be missing something, seems to easy

Reply:
Doesn't control for situations where we route multiple different DIPs, multiple account changes then will pass through with 0 routed qty potentially. That is a slightly different case than what we're trying to solve here though